File: VB/NetFramework/net48/09_GcComboBox/AutoComplete.vb
Imports System.Windows.Forms

Namespace _09_GcComboBox
	Public Partial Class AutoComplete
        Inherits InputManWin13_Demo.DemoBase_settings
        Public Sub New()
			InitializeComponent()
			InitializeForm()
		End Sub

		Private Sub InitializeForm()
			' xmlファイルからショートカット機能一覧取得
			Dim filePath As System.String = Application.StartupPath
			Me.dataSet1.ReadXml(filePath & "\Resources\XMLDataFile.xml")
			gcComboBox1.DataSource = dataSet1.Tables(0)

			' ***** 初期値
			gcComboBox1.AutoComplete.CandidateListItemFont = gcComboBox1.Font
			gcComboBox2.SelectedIndex = Convert.ToInt32(gcComboBox1.AutoComplete.MatchingMode)
			For i As Integer = 1 To gcComboBox1.ListColumns.Count - 1
				gcComboBox1.ListColumns(i).Visible = False
			Next

			gcComboBox1.ListColumns(0).Width = gcComboBox1.Width
			gcComboBox1.DropDown.Direction = GrapeCity.Win.Editors.DropDownDirection.BelowLeft
		End Sub

        Private Sub gcComboBox2_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs) Handles gcComboBox2.SelectedIndexChanged
            ' 検索モード
            gcComboBox1.AutoComplete.MatchingMode = CType(gcComboBox2.SelectedIndex, GrapeCity.Win.Editors.AutoCompleteMatchingMode)
        End Sub

        Private Sub checkBox1_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs) Handles checkBox1.CheckedChanged
            ' ハイライト表示
            gcComboBox1.AutoComplete.HighlightMatchedText = checkBox1.Checked
        End Sub

        Private Sub sideButton1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles sideButton1.Click
            ' 検索文字列のフォント
            Dim fd As New FontDialog()

            fd.Font = gcComboBox1.AutoComplete.HighlightStyle.Font
            fd.ShowColor = False

            If fd.ShowDialog() <> DialogResult.Cancel Then
                gcComboBox1.AutoComplete.HighlightStyle.Font = fd.Font
                gcTextBox1.Text = fd.Font.ToString()
            End If
        End Sub

        Private Sub gcColorPicker1_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs) Handles gcColorPicker1.SelectedIndexChanged
            ' 検索文字列の前景色
            gcComboBox1.AutoComplete.HighlightStyle.ForeColor = gcColorPicker1.SelectedColor
        End Sub

        Private Sub gcColorPicker2_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs) Handles gcColorPicker2.SelectedIndexChanged
            ' 検索文字列の背景色
            gcComboBox1.AutoComplete.HighlightStyle.BackColor = gcColorPicker2.SelectedColor
        End Sub

        Private Sub sideButton2_Click(ByVal sender As Object, ByVal e As EventArgs) Handles sideButton2.Click
            ' 候補リストのフォント
            Dim fd As New FontDialog()

            fd.Font = gcComboBox2.AutoComplete.CandidateListItemFont
            fd.ShowColor = False

            If fd.ShowDialog() <> DialogResult.Cancel Then
                gcComboBox1.AutoComplete.CandidateListItemFont = fd.Font
                gcTextBox2.Text = fd.Font.ToString()
            End If
        End Sub
	End Class
End Namespace

File: CS/NetFramework/net48/09_GcComboBox/AutoComplete.cs
using System;
using System.Windows.Forms;

namespace InputManWin13_Demo._09_GcComboBox
{
    public partial class AutoComplete : InputManWin13_Demo.DemoBase_settings
    {
        public AutoComplete()
        {
            InitializeComponent();
            InitializeForm();
        }

        private void InitializeForm()
        {
            // ***** 初期設定
            gcComboBox2.SelectedIndexChanged += new EventHandler(gcComboBox2_SelectedIndexChanged);
            sideButton1.Click += new EventHandler(sideButton1_Click);
            checkBox1.CheckedChanged += new EventHandler(checkBox1_CheckedChanged);
            gcColorPicker1.SelectedIndexChanged += new EventHandler(gcColorPicker1_SelectedIndexChanged);
            gcColorPicker2.SelectedIndexChanged += new EventHandler(gcColorPicker2_SelectedIndexChanged);
            sideButton2.Click += new EventHandler(sideButton2_Click);

            // xmlファイルからショートカット機能一覧取得
            System.String filePath = Application.StartupPath;
            this.dataSet1.ReadXml(filePath + "\\Resources\\XMLDataFile.xml");
            gcComboBox1.DataSource = dataSet1.Tables[0];

            // ***** 初期値
            gcComboBox1.AutoComplete.CandidateListItemFont = gcComboBox1.Font;
            gcComboBox2.SelectedIndex = Convert.ToInt32(gcComboBox1.AutoComplete.MatchingMode);
            for (int i = 1; i < gcComboBox1.ListColumns.Count; i++)
            {
                gcComboBox1.ListColumns[i].Visible = false;
            }

            gcComboBox1.ListColumns[0].Width = gcComboBox1.Width;
            gcComboBox1.DropDown.Direction = GrapeCity.Win.Editors.DropDownDirection.BelowLeft;
        }

        private void gcComboBox2_SelectedIndexChanged(object sender, EventArgs e)
        {
            // 検索モード
            gcComboBox1.AutoComplete.MatchingMode = (GrapeCity.Win.Editors.AutoCompleteMatchingMode)gcComboBox2.SelectedIndex;
        }

        private void checkBox1_CheckedChanged(object sender, EventArgs e)
        {
            // ハイライト表示
            gcComboBox1.AutoComplete.HighlightMatchedText = checkBox1.Checked;
        }

        private void sideButton1_Click(object sender, EventArgs e)
        {
            // 検索文字列のフォント
            FontDialog fd = new FontDialog();

            fd.Font = gcComboBox1.AutoComplete.HighlightStyle.Font;
            fd.ShowColor = false;

            if (fd.ShowDialog() != DialogResult.Cancel)
            {
                gcComboBox1.AutoComplete.HighlightStyle.Font = fd.Font;
                gcTextBox1.Text = fd.Font.ToString();
            }
        }

        private void gcColorPicker1_SelectedIndexChanged(object sender, EventArgs e)
        {
            // 検索文字列の前景色
            gcComboBox1.AutoComplete.HighlightStyle.ForeColor = gcColorPicker1.SelectedColor;
        }

        private void gcColorPicker2_SelectedIndexChanged(object sender, EventArgs e)
        {
            // 検索文字列の背景色
            gcComboBox1.AutoComplete.HighlightStyle.BackColor = gcColorPicker2.SelectedColor;
        }

        private void sideButton2_Click(object sender, EventArgs e)
        {
            // 候補リストのフォント
            FontDialog fd = new FontDialog();

            fd.Font = gcComboBox2.AutoComplete.CandidateListItemFont;
            fd.ShowColor = false;

            if (fd.ShowDialog() != DialogResult.Cancel)
            {
                gcComboBox1.AutoComplete.CandidateListItemFont = fd.Font;
                gcTextBox2.Text = fd.Font.ToString();
            }
        }
    }
}
